recipes/pybind11/all/conanfile.py

```diff
@@ -50,24 +43,39 @@ def package(self):
         self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
         cmake = self._configure_cmake()
         cmake.install()
-        lib_folder = os.path.join(
-            self.package_folder, "lib", "cmake", "pybind11")
-        os.rename(
-            os.path.join(lib_folder, "pybind11Config.cmake"),
-            os.path.join(lib_folder, "pybind11Install.cmake"))
-        os.unlink(os.path.join(lib_folder, "pybind11ConfigVersion.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Common.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11NewTools.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Targets.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Config.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11ConfigVersion.cmake"))
 
     def package_id(self):
         self.info.header_only()
 
     def package_info(self):
-        self.cpp_info.includedirs.append(os.path.join(
-            self.package_folder, "include", "pybind11"))
-
         cmake_base_path = os.path.join("lib", "cmake", "pybind11")
-        self.cpp_info.builddirs = [cmake_base_path]
+        if tools.Version(self.version) >= "2.6.0":
+            self.cpp_info.components["main"].names["cmake_find_package"] = "pybind11"
+            self.cpp_info.components["main"].builddirs = [cmake_base_path]
+            for generator in ["cmake_find_package", "cmake_find_package_multi"]:
+                self.cpp_info.components["main"].build_modules[generator].append(os.path.join(cmake_base_path, "FindPythonLibsNew.cmake"))
+                self.cpp_info.components["main"].build_modules[generator].append(os.path.join(cmake_base_path, "pybind11Tools.cmake"))
+            self.cpp_info.components["headers"].includedirs = [os.path.join("include", "pybind11")]
+            self.cpp_info.components["headers"].requires = ["main"]
+            self.cpp_info.components["embed"].requires = ["main"]
+            self.cpp_info.components["module"].requires = ["main"]
+            self.cpp_info.components["python_link_helper"].requires = ["main"]
+            self.cpp_info.components["windows_extras"].requires = ["main"]
+            self.cpp_info.components["lto"].requires = ["main"]
+            self.cpp_info.components["thin_lto"].requires = ["main"]
+            self.cpp_info.components["opt_size"].requires = ["main"]
+            self.cpp_info.components["python2_no_register"].requires = ["main"]
```

Comment:
those components names are too generic for pkg-config generator

Reply:
those are the names of the CMake targets in pybind11. Do you have any suggestions? I will resume work on this PR once Conan is updated and let's see if those components really make sense

Reply:
yes, add a prefix for conan component name, and set cmake_find_package and cmake_find_package_multi names

Reply:
Consider my comment https://github.com/conan-io/conan-center-index/pull/4445#issuecomment-821304460, I'd prefer to move this forward and then improve the components' related stuff.

Reply:
components are part of the interface (downstream recipes might rely in those conan components in their package_info). It's better to not introduce generic component names, because we would have to remove them later, and it would be a breaking change in the recipe.